app/Services/Validation/ApplicationTimelineValidator.php

```diff
@@ -0,0 +1,429 @@
+    {
+        // Field 'language_test_confirmed' needs to be validated conditionally.
+        return [
+            'language_requirement_confirmed' => ['required', 'boolean', 'accepted'],
+            'citizenship_declaration_id' => ['required', 'exists:citizenship_declarations,id'],
+            'veteran_status_id' => ['required', 'exists:veteran_statuses,id'],
+            'education_requirement_confirmed' => ['required', 'boolean', 'accepted'],
+        ];
+    }
+
+    /**
+     * Validator instance for the Basic Info step.
+     *
+     * @param JobApplication $application Job Application object.
+     *
+     * @return \Illuminate\Contracts\Validation\Validator
+     */
+    public function basicsValidator(JobApplication $application)
+    {
+        $application->loadMissing('job_poster');
+
+        // Check to see if this application is related to a Job Poster with
+        // a bilingual requirement.
+        $langRequirement = $application->job_poster->language_requirement_id;
+        $langTestRequirement = LanguageRequirement::where('name', 'bilingual_intermediate')
+            ->orWhere('name', 'bilingual_advanced')
+            ->pluck('id');
+
+        $validator = Validator::make($application->toArray(), $this->basicInfoSimpleRules());
+
+        // Conditionally check for 'language_test_confirmed' if the
+        // closure returns true.
+        $validator->sometimes(
+            'language_test_confirmed',
+            'required|boolean|accepted',
+            function ($input) use ($langRequirement, $langTestRequirement) {
+                return in_array($langRequirement, $langTestRequirement->toArray());
+            }
+        );
+
+        return $validator;
+    }
+
+    /**
+     * Helper function to return completeness for the Basic Info step.
+     *
+     * @param JobApplication $application Job Application object.
+     *
+     * @return boolean
+     */
+    public function basicsComplete(JobApplication $application)
+    {
+        $validator = $this->basicsValidator($application);
+        return $validator->passes();
+    }
+
+    /* End Step 1 --------------------------------------------------------------------------------------------------- */
+
+    /* Step 2 ------------------------------------------------------------------------------------------------------- */
+
+    /**
+     * Validator instance for the Experience step.
+     *
+     * @param JobApplication $application Job Application object.
+     *
+     * @return \Illuminate\Contracts\Validation\Validator
+     */
+    public function experienceValidator(JobApplication $application)
+    {
+        $application->loadMissing('job_poster', 'job_poster.criteria');
+
+        $essentialCriteriaType = CriteriaType::where('name', 'essential')->first()->id;
+        $hardSkillType = SkillType::where('name', 'hard')->first()->id;
+
+        $jobCriteria = $application->job_poster->criteria;
+        $requiredCriteriaSkillIds = $jobCriteria
```

Comment:
I renamed requiredCriteria to requiredCriteriaSkillIds, and a couple other similar changes, to better reflect what they are now.